### PR TITLE
Allow third-party services to listen to openURL on iOS (needed for Google Sign-In)

### PR DESCRIPTION
**Pull request by collinjackson**

### CODE REVIEW

**Comment by Hixie** [comment]
What's the current plan for moving this stuff into a different repo?

**Comment by collinjackson** [comment]
Not done yet, still planned.

**Comment by Hixie** [comment]
But what's the plan? Is there anything blocking us doing it?

**Comment by collinjackson** [comment]
The build infrastructure isn't currently set up for it, and we're still figuring out our approach to third-party services so we haven't been doing implementation work that we might need to throw away.

I filed https://github.com/flutter/flutter/issues/2773 to track this.

File: sky/services/google/ios/google_sign_in_impl.mm [hunk 1]
```diff
@@ -0,0 +1,133 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "sky/services/google/ios/google_sign_in_impl.h"
+#include "base/logging.h"
+#include "base/strings/sys_string_conversions.h"
+
+#import <GoogleSignIn.h>
```
Comment (abarth) [inline sky/services/google/ios/google_sign_in_impl.mm]: Where does this header come from?

**Comment by abarth** [comment]
I think the biggest question here is whether we want to have something so Google-specific in the repo.  The ideal thing would be to be able to build these plugins on Travis or something so that it would be clear how sign-in services could become peers.  I understand that we don't want this worked to be blocked behind that...  Maybe we should do what we did for GCM and put the code in its own repo and then use DEPS to pull it into the engine build.  It amounts to the same thing but at least it shows more of the intent that these be separate vendor-specific pieces.

**Comment by collinjackson** [comment]
I've moved the code that's specific to Google Sign In into a separate repo in flutter/google_sign_in#2

PTAL

File: sky/shell/platform/ios/FlutterAppDelegate.mm [hunk 2]
```diff
@@ -58,4 +58,26 @@ - (BOOL)application:(UIApplication*)application
...
+    @"url": url,
+    @"sourceApplication": sourceApplication,
+  } mutableCopy];
+  if (annotation != nil)
+    [dict setValue:annotation forKey:@"annotation"];
+  [[NSNotificationCenter defaultCenter] postNotificationName:@"openURL"
+                                                      object:self
+                                                    userInfo:dict];
+  BOOL handled = ((NSNumber *)dict[@"handled"][@"value"]).boolValue;
+  NSLog(@"Handled: %d", handled);
```
Comment (abarth) [inline sky/shell/platform/ios/FlutterAppDelegate.mm]: Do we want these log statements in production?

**Comment by abarth** [comment]
LGTM.  Consider if @chinmaygarde should look as well.

**Pull request merged**